tasks/main.yml

```diff
@@ -22,32 +22,35 @@
 - name: Make sure Docker CE is the version specified.
   apt:
     name: "docker-ce={{ docker_version }}"
-    state: present
+    state: "{{ docker_package_state }}"
     force: true
     install_recommends: "{{ docker_install_recommends }}"
 
-# Shim for Docker 18.09.0 on Debian Stretch.
-- import_tasks: docker-1809-shim.yml
-  when:
-    - ansible_service_mgr == 'systemd'
-    - "'18.09.0' in docker_version"
+- name: process docker
```

Comment:
Sorry for this.

I've moved some tasks (`docker-1809-shim.yml`, `Ensure Docker is started` and `Ensure docker users are added to the docker group`) into a block to avoid them when uninstalling docker. 

IMHO: these tasks shouldn't be executed if there is no docker.